atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/FastFailoverProxy.java

```diff
@@ -41,7 +41,7 @@
  * leader election is still taking place.
  */
 public final class FastFailoverProxy<T> extends AbstractInvocationHandler {
-    private static final Duration TIME_LIMIT = Duration.ofSeconds(10);
+    private static final Duration TIME_LIMIT = Duration.ofSeconds(15);
```

Comment:
Can we add a metric and alerting on when this deadline is exceeded?

Reply:
does something in `ClientOptions` need to be changed also? if so @jeremyk-91, we really need to centralise this stuff, otherwise it's too easy to break assumptions here

Reply:
Adding metrics/alerting makes sense, though I would not block this on that given the issue recurred and caused another batch of errors today - as long as we're tracking it somewhere.

ClientOptions: good flag. I'm guessing you're worried on `SocketTimeoutException`? Those get wrapped and become retryable, so if we were facing that we would fire one off, get blocked for 10 seconds, and if that's bad because we haven't had 15 seconds fire off another one (which would again get blocked for 10 seconds). If that's the only case then that should be fine, though I might have missed something else.

Reply:
Don't want to block, but I'd like this to be the next change, and it's tracked. @Jolyon-S can you make sure we track this?

Reply:
I will add an issue and make sure we track it to completion

Reply:
@jeremyk-91 I can't see any reference to 10s in `ClientOptions` - is this in a related class?

Reply:
not 10s, but regards to timeouts in general and how they interplay. This will only retry when 308s occur i.e. you've been speaking to the non-leader or you're in the middle of a leader election as opposed to "we're not getting anything from the socket" for which we won't retry if that limit is reached.

At the very least, having this constant being in `ClientOptions` and a comment on what it represents and how it interplays with some of the other constants in that class would be good here.